src/www/ui/api/Controllers/LicenseController.php

```diff
@@ -463,4 +463,49 @@ public function deleteAdminLicenseCandidate($request, $response, $args)
     }
     return $response->withJson($resInfo->getArray(), $resInfo->getCode());
   }
+
+  /**
+   * Get suggested license from reference text
+   *
+   * @param Request $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function getSuggestedLicense($request, $response, $args)
+  {
+    $body =  $this->getParsedBody($request);
+    $rfText = $body["referenceText"];
+    if (!Auth::isAdmin()) {
+      $resInfo = new Info(403, "Only admin can perform this operation.",
+        InfoType::ERROR);
+      return $response->withJson($resInfo->getArray(), $resInfo->getCode());
+    }
+    if (empty($rfText)) {
+      $error = new Info(400, "Reference text is missing.", InfoType::ERROR);
+      return $response->withJson($error->getArray(), $error->getCode());
+    }
+    $adminLicenseCandidate = $this->restHelper->getPlugin('admin_license_candidate');
+
+    list ($suggestIds, $rendered) = $adminLicenseCandidate->suggestLicenseId($rfText);
+
+    if (! empty($suggestIds)) {
+      $suggest = $suggestIds[0];
+      $suggestLicense = $adminLicenseCandidate->getDataRow($suggest, 'ONLY license_ref');
+      $suggestLicense = [
+        'id' => intval($suggestLicense['rf_pk']),
+        'spdxName' => $suggestLicense['rf_spdx_id'],
+        'shortName' => $suggestLicense['rf_shortname'],
+        'fullName' => $suggestLicense['rf_fullname'],
+        'text' => $suggestLicense['rf_text'],
```

Comment:
Would even suggest to return raw Highlight info at following location if the call is coming from REST API.
https://github.com/fossology/fossology/blob/f2650a9de30fcb2bb58cad3fd1caff305671ce1b/src/monk/ui/oneshot.php#L86

It can very similarly be used as the [HighlightInfo](https://github.com/fossology/fossology/blob/master/src/www/ui/api/documentation/openapi.yaml#L4683) model used by `GET /uploads/{id}/item/{itemId}/highlight`

Reply:
The $rendered variable has some extra information like span tags that I think are not appropriate for the output, as we did on the previous similar tasks. I think we can return only the licenseText of the suggested license, without any modifications and also the associated array of the highlight entries. How does this sounds @GMishx ?